pkg/mcs/exportspec_test.go

```diff
@@ -40,34 +41,91 @@ func TestMarshalUnmarshal(t *testing.T) {
 		expected *mcs.ExportSpec
 		mutate   func(mcs.ExportSpec) *mcs.ExportSpec
 	}{
-		/*
-			"empty": {},
-			"missing: timestamp": {},
-			"missing: cluster": {},
-			"missing: namespace": {},
-			"missing: name": {},
-			"missing: service":       {},
-			"missing: service.type":  {},
-			"missing: service.ports": {},
-			"simple spec": {},
-			"multiple ports":         {},
-		*/
+
+		"empty": {
+			expected: &mcs.ExportSpec{},
+			mutate:   func(mcs.ExportSpec) *mcs.ExportSpec { return &mcs.ExportSpec{} },
+		},
+
+		"missing: timestamp": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.CreatedAt = metav1.Unix(0, 0)
+				return &es
+			},
+		},
+
+		"missing: cluster": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.ClusterID = ""
+				return &es
+			},
+		},
+
+		"missing: namespace": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Namespace = ""
+				return &es
+			},
+		},
+
+		"missing: name": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Name = ""
+				return &es
+			},
+		},
+
+		"missing: service": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Service = mcs.GlobalProperties{}
+				return &es
+			},
+		},
+
+		"missing: service.type": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Service.Type = ""
+				return &es
+			},
+		},
+
+		"missing: service.ports": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Service.Ports = []mcsv1a1.ServicePort{}
+				return &es
+			},
+		},
+
+		"simple spec": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				return &es
+			},
+		},
+
+		"multiple ports": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				new_port_81 := mcsv1a1.ServicePort{Port: 81, Name: "http2", Protocol: corev1.ProtocolTCP}
+				new_port_82 := mcsv1a1.ServicePort{Port: 82, Name: "http3", Protocol: corev1.ProtocolTCP}
+				es.Service.Ports = append(es.Service.Ports, new_port_81)
```

Comment:
it's fine as is, but for reference, Go does let you do this inline with:
```go 
   es.Service.Ports = append(es.Service.Ports, mcsv1a1.ServicePort{
      Port: 81, 
      Name: "http2", 
      Protocol: corev1.ProtocolTCP,
      })
```

which saves on an intermediate variable for the garbage collector